### ISSUE TITLE
when I trained on my own dataset had gotten a great result on val_set(about loss :0.05 and mAP 0.97), but get a bad result on test_set

### ISSUE THREAD

**Issue by vincezengqiang** [fizyr/keras-retinanet#762]
I prepare a train set and val set to train the net, and get the message that the loss decrease to 0.05(the regression loss is about 0.048), and the evaluate message of mAP after the last epoch is about 0.96.
And  then i convert the train model to inference model, and run the net on my test set, the result is so bad.I even try to test the net on the val set which used to evaluate during the train, get a bad result either.
Could you please give me some advices? 
Sincerely!

**Comment by JerryIndus** [comment]
Hello, I meet the same problem, and want to ask you if you have already solve this problem?

**Comment by MsRabbits** [comment]
> I prepare a train set and val set to train the net, and get the message that the loss decrease to 0.05(the regression loss is about 0.048), and the evaluate message of mAP after the last epoch is about 0.96.
> And then i convert the train model to inference model, and run the net on my test set, the result is so bad.I even try to test the net on the val set which used to evaluate during the train, get a bad result either.
> Could you please give me some advices?
> Sincerely!

hey dude, can you tell me how to evaluate the model on a new test set after getting the inference model? thx!
